diff --git a/modules/hdbt_admin_editorial/hdbt_admin_editorial.module b/modules/hdbt_admin_editorial/hdbt_admin_editorial.module
index 7494c57..72ad075 100644
--- a/modules/hdbt_admin_editorial/hdbt_admin_editorial.module
+++ b/modules/hdbt_admin_editorial/hdbt_admin_editorial.module
@@ -349,6 +349,13 @@ function hdbt_admin_editorial_form_linkit_editor_dialog_form_alter(&$form, FormS
 function _hdbt_admin_editorial_attributes_validate(array &$form, FormStateInterface $form_state) {
   $attributes = $form_state->getValue('attributes');

+  if ($attributes['target']) {
+    $form_state->setValue(['attributes', 'target'], '_blank');
+  }
+  else {
+    $form_state->setValue(['attributes', 'target'], '_self');
+  }
+
   // Allow icons only for the links with button design.
   if (isset($attributes['data-selected-icon'])) {
     if ($attributes['data-design'] === 'link' || empty($attributes['data-selected-icon'])) {
